crates/vizia_core/src/views/svg_display.rs

```diff
@@ -0,0 +1,189 @@
+use std::cell::RefCell;
+
+use femtovg::{Paint, Path};
+use usvg::{LineCap, LineJoin, NodeKind, PathSegment};
+
+use crate::{prelude::*, vg};
+
+/// A svg display used to display vector graphics to the screen.
+///
+/// # Examples
+///
+/// ## Basic Svg Display
+///
+/// A svg display can be used to simply load and display vector graphics on the screen.
+///
+/// ```ignore
+/// # use vizia_core::prelude::*;
+/// #
+/// # let cx = &mut Context::default();
+/// #
+/// SvgDisplay::new(cx, SvgTree::from_str(include_str!("path/to/my/graphic.svg"), &SvgOptions::default(), None));
+/// ```
+///
+/// ## Svg Display with overriden color
+///
+/// Displaying a Svg where the color is other than from the given svg.
+///
+/// ```ignore
+/// # use vizia_core::prelude::*;
+/// #
+/// # let cx = &mut Context::default();
+/// #
+/// SvgDisplay::new(cx, SvgTree::from_str(include_str!("path/to/my/graphic.svg"), &SvgOptions::default(), Some(Color::red())));
+/// ```
+pub struct SvgDisplay {
+    tree: RefCell<SvgTree>,
+}
+
+enum SvgEvent {
+    TreeChanged(SvgTree),
+}
+
+impl SvgDisplay {
+    /// Creates a new svg display.
+    ///
+    /// # Examples
+    ///
+    /// ```ignore
+    /// # use vizia_core::prelude::*;
+    /// #
+    /// # let cx = &mut Context::default();
+    /// #
+    /// SvgDisplay::new(cx, SvgTree::from_str(include_str!("path/to/my/graphic.svg"), &SvgOptions::default(), None));
+    /// ```
+    pub fn new<'a>(cx: &'a mut Context, svg_tree: impl Res<SvgTree>) -> Handle<'a, Self> {
+        Self { tree: RefCell::new(SvgTree::default()) }.build(cx, |cx| {
+            svg_tree.set_or_bind(cx, cx.current(), |cx, _entity, val| {
+                cx.emit(SvgEvent::TreeChanged(val));
+            });
+        })
+    }
+}
+
+impl View for SvgDisplay {
+    fn element(&self) -> Option<&'static str> {
+        Some("svg_display")
+    }
+
+    fn event(&mut self, _cx: &mut EventContext, event: &mut Event) {
+        if let Some(event) = event.take::<SvgEvent>() {
+            match event {
+                SvgEvent::TreeChanged(new_tree) => {
+                    self.tree.replace(new_tree);
+                }
+            }
+        }
```

Comment:
This could be
```suggestion
        if let Some(SvgEvent::TreeChanged(new_tree)) = event.take::<SvgEvent>() {
            self.tree.replace(new_tree);
        }
```